src/main/scala/mesosphere/marathon/api/akkahttp/v2/EventsController.scala

```diff
@@ -49,19 +49,20 @@ class EventsController(
     */
   def eventsSSE(): Route = {
     def isAllowed(allowed: Set[String])(event: MarathonEvent): Boolean = allowed.isEmpty || allowed(event.eventType)
+
     authenticated.apply { implicit identity =>
       authorized(ViewResource, AuthorizedResource.Events).apply {
-        parameters('event_type.*) { events =>
-          extractClientIP { clientIp =>
-            complete {
-              Source
-                .fromGraph[MarathonEvent, NotUsed](
-                  new EventStreamSourceGraph(eventBus, conf.eventStreamMaxOutstandingMessages(), clientIp)
-                )
-                .filter(isAllowed(events.toSet))
-                .map(event => ServerSentEvent(`type` = event.eventType, data = Json.stringify(Formats.eventToJson(event))))
-                .keepAlive(5.second, () => ServerSentEvent.heartbeat)
-            }
+        (parameters('event_type.*, "plan-format".?) & extractClientIP) { (events, light, clientIp) =>
+          complete {
+            Source
+              .fromGraph[MarathonEvent, NotUsed](
+                new EventStreamSourceGraph(eventBus, conf.eventStreamMaxOutstandingMessages(), clientIp)
+              )
+              .filter(isAllowed(events.toSet))
+              .map(event => ServerSentEvent(`type` = event.eventType, data = Json.stringify(
+                Formats.eventToJson(event, light.exists(_.contentEquals("light")))
```

Comment:
I've only skimmed this PR but wouldn't it be lighter to explicitly pass a `Writes` in the controller instead of passing down the config into the formats definition? Something like (watch out, imperative pseudo code 😱 )
```
val writes = if (light) lightWrites else heavyWrites
toJson(event)(writes)
```
.. where `lightWrites` contains writes for the given events using light raml generated formats, and `heavyWrites` uses the heavy formats. Passing down the flag seems kind of 🐟 'y.

So instead of importing `mesosphere.marathon.api.v2.json.Formats`, you'd explicitly reference the format you want.